File: sort/select_sort.cpp
#include <iostream>
#include <vector>
using namespace std;

void printVector(const vector<int>& nums) {
    for (const int& num : nums) {
        cout << num << " ";
    }
    cout << endl;
}

void selectSort(vector<int>& nums) {
    for (size_t i{0UL}; i < nums.size() - 1; i++) {
        int min_index = i;
        for (size_t j = i + 1; j < nums.size(); j++) {
            if (nums[j] < nums[min_index]) {
                min_index = j;
            }
        }
        swap(nums[i], nums[min_index]);
        printVector(nums);
    }
}

int main(int argc, char const *argv[])
{
    std::vector<int> nums{2, 3, 1, 5, 4, 7, 6};
    cout << "Original nums: ";
    printVector(nums);
    selectSort(nums);
    cout << "After Sorted: ";
    printVector(nums);
    return 0;
}

File: sort/select_sort.py
def seletSort(nums: list):
    n = len(nums)
    for i in range(0, n):
        min_index = i
        for j in range(i + 1, n):
            if (nums[j] < nums[min_index]):
                min_index = j
        # nums[i] = nums[min_index]
        nums[i], nums[min_index] = nums[min_index], nums[i]
        print(nums)

def selectSort2(nums: list):
    n = len(nums)
    for i in range(0, n - 1):
        for j in range(i + 1, n):
            if (nums[j] < nums[i]):
                nums[i], nums[j] = nums[j], nums[i]
        print(nums)

if __name__ == "__main__":
    nums = [10, 3, 1, 8, 4,  7, 6]
    print(nums)
    print("After Sort:")
    # seletSort(nums)
    selectSort2(nums)
    # print(1 in (1, 3, 4, 5))
    # print(nums)
